### PR TITLE
ci(action): add scorecard automation

### PR DESCRIPTION
**Pull request by v1v**
As explained in https://github.com/ossf/scorecard-action#installation

## What does this PR do?

Use scorecard to run some analysis for this project

## Why is it important?

Help to see how well the integration works

### CODE REVIEW

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
```
Comment (v1v) [inline .github/workflows/scorecard.yml:1]:
```suggestion
---
# This workflow uses actions that are not certified by GitHub. They are provided
```

File: .github/workflows/scorecard.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  workflow_dispatch: ~
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule: ~
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '38 12 * * 0'
+  push:
+    branches: [ "main" ]
```
Comment (v1v) [inline .github/workflows/scorecard.yml:17]:
```suggestion
    branches: ["main"]
```

File: .github/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
```
Comment (v1v) [inline .github/workflows/scorecard.yml:33]:
```suggestion
```

File: .github/workflows/scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@e38b1902ae4f44df626f11ba0734b14fb91f8f86 # v2.1.2
```
Comment (v1v) [inline .github/workflows/scorecard.yml:42]:
```suggestion
        uses: ossf/scorecard-action@e38b1902ae4f44df626f11ba0734b14fb91f8f86  # v2.1.2
```

File: .github/workflows/scorecard.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
```
Comment (v1v) [inline .github/workflows/scorecard.yml:37]:
```suggestion
        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8  # v3.1.0
```

File: .github/workflows/scorecard.yml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action#authentication-with-pat.
+          # repo_token: ${{ secrets.SCORECARD_TOKEN }}
+
+          # Public repositories:
+          #   - Publish results to OpenSSF REST API for easy access by consumers
+          #   - Allows the repository to include the Scorecard badge.
+          #   - See https://github.com/ossf/scorecard-action#publishing-results.
+          # For private repositories:
+          #   - `publish_results` will always be set to `false`, regardless
+          #     of the value entered here.
```
Comment (v1v) [inline .github/workflows/scorecard.yml:58]:
```suggestion
```

File: .github/workflows/scorecard.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+          publish_results: true
+
+      # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
+      # format to the repository Actions tab.
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
```
Comment (v1v) [inline .github/workflows/scorecard.yml:64]:
```suggestion
        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8  # v3.1.0
```

File: .github/workflows/scorecard.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        with:
+          name: SARIF file
+          path: results.sarif
+          retention-days: 5
+
+      # Upload the results to GitHub's code scanning dashboard.
+      - name: "Upload to code-scanning"
+        uses: github/codeql-action/upload-sarif@17573ee1cc1b9d061760f3a006fc4aac4f944fd5 # v2.2.4
```
Comment (v1v) [inline .github/workflows/scorecard.yml:72]:
```suggestion
        uses: github/codeql-action/upload-sarif@17573ee1cc1b9d061760f3a006fc4aac4f944fd5  # v2.2.4
```

File: .github/workflows/scorecard.yml [hunk 9]
```diff
@@ -0,0 +1,74 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
```
Comment (v1v) [inline .github/workflows/scorecard.yml:3]:
```suggestion
```

File: .github/workflows/scorecard.yml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8  # v3.1.0
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@e38b1902ae4f44df626f11ba0734b14fb91f8f86  # v2.1.2
```
Comment (amannocci) [inline .github/workflows/scorecard.yml:37]:
```suggestion
        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736  # v2.3.1
```

File: .github/workflows/scorecard.yml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          publish_results: true
+
+      # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
+      # format to the repository Actions tab.
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8  # v3.1.0
```
Comment (amannocci) [inline .github/workflows/scorecard.yml:46]:
```suggestion
        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32  # v3.1.3
```

File: .github/workflows/scorecard.yml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          name: SARIF file
+          path: results.sarif
+          retention-days: 5
+
+      # Upload the results to GitHub's code scanning dashboard.
+      - name: "Upload to code-scanning"
+        uses: github/codeql-action/upload-sarif@17573ee1cc1b9d061760f3a006fc4aac4f944fd5  # v2.2.4
```
Comment (amannocci) [inline .github/workflows/scorecard.yml:54]:
```suggestion
        uses: github/codeql-action/upload-sarif@74483a38d39275f33fcff5f35b679b5ca4a26a99  # v2.22.5
```

**Review by amannocci — CHANGES_REQUESTED** [review]

File: .github/workflows/scorecard.yml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8  # v3.1.0
```
[the remark this answers is not in the document]
Comment (v1v) [reply .github/workflows/scorecard.yml:32]: Good finding, somehow the workflow UI automation to create this through the UI is not up-to-date :/

**Review by amannocci — APPROVED** [review]
LGTM

**Pull request merged**